connectors/preflight_check.py

```diff
@@ -69,3 +71,26 @@ async def _check_system_indices_with_retries(self):
                     attempts += 1
                     await self._sleeps.sleep(self.preflight_idle)
         return False
+
+    def _check_configuration_validity(self):
+        # "Native" mode
+        configured_native_types = "native_service_types" in self.config
+        force_allowed_native = self.config.get("_force_allow_native", False)
+        if configured_native_types and not force_allowed_native:
+            logger.warning(
+                "The configuration 'native_service_types' has been deprecated. Please remove this configuration."
+            )
+            logger.warning(
+                "Native Connectors are only supported internal to Elastic Cloud deployments, which this process is not."
+            )
+
+        # Connector client mode
+        configured_connector_id = self.config.get("connector_id", None)
+        configred_service_type = self.config.get("service_type", None)
+        if (
+            not (configured_connector_id and configred_service_type)
+            and not force_allowed_native
```

Comment:
Should also check whether there's value in `native_service_types`? `force_allowed_native` doesn't mean it runs in native mode.

Reply:
I want this printed if a customer is running in native mode on self-managed infrastructure. I want it to not only tell them not to configure that, but to tell them what _to_ configure. Otherwise, they have to get errors twice:
1. configured to run native mode
1. get warning to remove `native_service_types`
1. remove `native_service_types` and restart
1. get warning that you have to configure `service_type` and `connector_id`
1. be frustrated that I didn't tell you that to start with.

I want to avoid that last bit of frustration. I also wanted to cover the case where they have both `native_service_types` AND `connector_id`/`service_type` configured, hence why these logs are in two conditionals, and not just one.